AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -75,7 +81,10 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
 
     id<ASLayoutable> finalLayoutable = [child finalLayoutable];
     if (finalLayoutable != child) {
-      [finalLayoutable.layoutOptions copyFromOptions:child.layoutOptions];
+      // Copy layout options
+      ASEnvironmentState environmentState = finalLayoutable.environmentState;
+      environmentState.layoutOptionsState = child.environmentState.layoutOptionsState;
```

Comment:
I am not completely certain, but I believe we should perform a merge operation here, treating the final layoutable as though it were a parent and merging upward the layout options from the current child.

I think this should have been even in the case of iterating through multiple children, each time a final layoutable is created, check if the object returned from final layoutable is different then the object is called on, then perform an upward merge.

The reason is that there may be properties specified on the final layout double that are specifically intended to override properties of the object it is substituting in for.